error_handling_on_chain/README.md

```diff
@@ -0,0 +1,127 @@
+contract in the
+[Hello World - Bringing A Blocky AS Function Call Attestation On Chain](../../hello_world_on_chain)
+example.
+
+In this example, we go a step further and show you how to extract the `Result`
+struct from the `successFunc` and `errorFunc` function call attestations in the
+[Error Handling](../error_handling) example.
+
+In [`contracts/User.sol`](contracts/User.sol), we define a `parseResult` function:
+
+```solidity
+function parseResult(
+    string memory resultString
+) public
+{
+    JsmnSolLib.Token[] memory tokens;
+    uint number;
+    uint success;
+    (success, tokens, number) = JsmnSolLib.parse(resultString, 50);
+
+    bool resultSuccess;
+    string memory resultError;
+    string memory valueString;
+
+    for (uint i = 0; i < number; i++) {
+        if (tokens[i].jsmnType == JsmnSolLib.JsmnType.STRING) {
+            string memory key = JsmnSolLib.getBytes(
+                resultString,
+                tokens[i].start,
+                tokens[i].end
+            );
+
+            if (keccak256(bytes(key)) == keccak256("Success")) {
+                resultSuccess = JsmnSolLib.parseBool(
+                    JsmnSolLib.getBytes(
+                        resultString,
+                        tokens[i + 1].start,
+                        tokens[i + 1].end
+                    )
+                );
+            } else if (keccak256(bytes(key)) == keccak256("Error")) {
+                resultError = JsmnSolLib.getBytes(
+                    resultString,
+                    tokens[i + 1].start,
+                    tokens[i + 1].end
+                );
+            } else if (keccak256(bytes(key)) == keccak256("Value")) {
+                valueString = JsmnSolLib.getBytes(
+                    resultString,
+                    tokens[i + 1].start,
+                    tokens[i + 1].end
+                );
+            }
+        }
+    }
+
+    console.log("\tSuccess: %s", resultSuccess);
+    console.log("\tError: %s", resultError);
+    console.log("\tValue: %s", valueString);
+
+    require(resultSuccess, resultError);
+    emit ResultValue(valueString);
+}
+```
+
+which takes in the verified `TAParserLib.FnCallClaims` `claims.Output` and uses
+`JsmnSolLib` to parse the JSON serialized `Result` struct. JSON serialization
+does not guarantee struct field order, so we loop over the tokens to extract the
+fields: `Result.Success`, `Result.Error`, and `Result.Value`. If `Result.Success`
+is `false`, we revert the transaction with `Result.Error` as the error message.
+If `Result.Success` is `true`, we emit `Result.Value` as an event, where
+`Result.Value` itself may be a JSON serialized struct representing the output
+of the function call.
+
+## Next steps
+
+Now that you have successfully run the example, you can start modifying it to
+fit your own needs. For instance, write a function that extracts the`Output`
+struct from the `ResultValue(valueString)` event. Check out other examples in
```

Comment:
I really like this suggestion. However, I think you want to suggest that they extract the `Output` struct from the `valueString` variable, not the event. Additionally, we don't have a well set up context here for what is the `Output` struct. Maybe add something to say that you're referring to the output struct defined/used in the `successFunc` of the error handling example.

Reply:
Done